introduce labels for test execution (#9284)
kyma-project/kyma: tests/end-to-end/upgrade/README.md

Run end-to-end upgrade tests for the Kyma [upgrade plan](https://github.com/kyma-project/test-infra/blob/master/prow/scripts/cluster-integration/kyma-gke-upgrade.sh) on Prow. The continuous integration flow looks as follows:

1. Install Kyma from the [latest](https://github.com/kyma-project/kyma/releases/latest) release.
2. Install the upgrade-test [helm chart](chart/upgrade). 

> 3. Execute all testdefinitions with the label `kyma-project.io/test.before-upgrade`.

Reviewer: ```suggestion
3. Execute all TestDefinitions with the label `kyma-project.io/test.before-upgrade`.
```

> 4. Upgrade Kyma cluster.

Reviewer: ```suggestion
4. Upgrade the Kyma cluster.
```

> 5. Execute all testdefinitions with the label `kyma-project.io/test.after-upgrade`.

Reviewer: ```suggestion
5. Execute all TestDefinitions with the label `kyma-project.io/test.after-upgrade`.
```

## Development

> >**NOTE:** The following approach for adding new tests to the upgrade scenario is not needed anymore. To prepare data during the preparation phase, create a new TestDefinition that executes your preparation code and add the label `kyma-project.io/before-upgrade=true`. To execute evaluation code during the evaluation phase, create a new TestDefinition with the label `kyma-project.io/after-upgrade=true` 

Reviewer 2: This section is about how to implement new upgrade test - for example to a new domain. I don't understand why you wrote the first sentence in the note.  This section describes how to implement the test, the interface which must be implemented, where the code should be created (which folder).

Author: I wrote this, because in order to implement a new upgrade test you don't have to shoehorn it into the existing upgrade test anymore. as long as you label the testdefinition correctly you are totally free to implement the test however you like. This makes it easier to introduce your own dependencies as you are not longer limited by dependencies of other tests (eg knative eventing 0.12 limits the version of the kubernetes client)

## Development

> >**NOTE:** The following approach for adding new tests to the upgrade scenario is not needed anymore. To prepare data during the preparation phase just create a new testdefinition that executes your preparation code and add the label `kyma-project.io/before-upgrade=true`. To execute evaluation code during the evaluation phase create a new testdefinition with the label `kyma-project.io/after-upgrade=true` 

Reviewer: ```suggestion
```

Reviewer: It seems like the exact same note is repeated 2 lines further. Maybe we only one of them would be enough. Since the note is about "adding new tests to the upgrade scenario", I think it should be under the "Add a new test" title.

This section presents how to add and run a new test. It also describes how to verify the code and ensure that your test is correct.

### Add a new test (Deprecated)

> >**DEPRECATION NOTICE:** The following approach for adding new tests to the upgrade scenario is not needed anymore. To prepare data during the preparation phase just create a new testdefinition that executes your preparation code and add the label `kyma-project.io/before-upgrade=true`. To execute evaluation code during the evaluation phase create a new testdefinition with the label `kyma-project.io/after-upgrade=true` 

Reviewer: ```suggestion
>**NOTE:** The following approach for adding new tests to the upgrade scenario is not needed anymore. To prepare data during the preparation phase, create a new TestDefinition that executes your preparation code and add the label `kyma-project.io/before-upgrade=true`. To execute evaluation code during the evaluation phase, create a new TestDefinition with the label `kyma-project.io/after-upgrade=true` 
```
